src/ripple/app/paths/Pathfinder.cpp

```diff
@@ -181,33 +199,23 @@ Pathfinder::Pathfinder (
         mDstAmount (saDstAmount),
         mSrcCurrency (uSrcCurrency),
         mSrcIssuer (uSrcIssuer),
-        mSrcAmount ({uSrcCurrency, uSrcIssuer}, 1u, 0, true),
+        convert_all_ (mDstAmount ==
+            STAmount (mDstAmount.issue (), 1u, 0, true)),
         mLedger (cache->getLedger ()),
         mRLCache (cache)
 {
-    assert (isXRP(uSrcCurrency) == isXRP(uSrcIssuer));
-}
+    if (! srcAmount)
+    {
+        mSrcAmount = STAmount({ uSrcCurrency,
```

Comment:
So I still don't fully understand this - by "fully" I mean, "Can I reproduce this without looking?" and I've been looking at this code in various incarnations for a long time now.

I can explain what happens to the variables - but what does this code mean in terms of business logic?